### PR TITLE
Add native GLFW binary builds

### LINKED ISSUE #727: .NET 6 arm64 builds
yeah a native arm64 build would be nice i guess... i cant get it to work myself and i think it attract a wide audience imo

Tracking:

- [x] Assimp - OSX arm64
- [x] Assimp - Win arm64
- [x] Assimp - Linux arm64
- [ ] DXVK - Win arm64
- [ ] DXVK - Linux arm64
- [x] DirectStorage - Win arm64
- [x] GLFW - OSX arm64
- [x] GLFW - Win arm64 
- [x] GLFW - Linux arm64 
- [x] MoltenVK
- [x] OpenAL Soft - OSX arm64
- [x] OpenAL Soft - Win arm64
- [x] OpenAL Soft - Linux arm64
- [ ] ANGLE - OSX arm64
- [ ] ANGLE - Win arm64
- [ ] ANGLE - Linux arm64
- [x] SDL - OSX arm64
- [x] SDL - Win arm64
- [x] SDL - Linux arm64
- [x] SPIR-V Cross - Linux arm64
- [x] SPIR-V Cross - OSX arm64
- [x] SPIR-V Cross - Win arm64
- [x] SPIR-V Reflect - Linux arm64
- [x] SPIR-V Reflect - OSX arm64
- [x] SPIR-V Reflect - Win arm64 
- [x] Shaderc - Linux arm64
- [x] Shaderc - OSX arm64
- [x] Shaderc - Win arm64 
- [ ] Vkd3d - Linux arm64 
- [x] Vulkan Loader - OSX arm64
- [ ] Vulkan Loader - Win arm64
- [x] Vulkan Loader - Linux arm64
- [x] SwiftShader - OSX arm64
- [ ] SwiftShader - Win arm64
- [x] SwiftShader - Linux arm64
- [x] WGPU - OSX arm64
- [x] WGPU - Win arm64
- [x] WGPU - Linux arm64

### PR DESCRIPTION
**Pull request by roeyskoe**
# Summary of the PR
Self building GLFW for Windows, Mac and Linux, including Mac arm64.
This partially resolves #727.

# Related issues, Discord discussions, or proposals
https://discord.com/channels/521092042781229087/587346162802229298/936229014249562172

# Further work
Currently the mac libraries cause a security warning, since they are not signed. This needs to be resolved before shipping them.

Silk is also still missing arm64 build for OpenAL, and ImGUI.NET also does not ship with arm natives.

### CODE REVIEW

File: .github/workflows/GLFW.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: glfw
```
Comment (Perksey) [inline .github/workflows/GLFW.yml:1]:
```suggestion
name: GLFW
```

File: build/nuke/Build.Native.cs [hunk 2]
```diff
@@ -99,4 +99,73 @@ string AndroidHome
...
+                    {
+                        InheritedShell($"{prepare} -DCMAKE_SYSTEM_PROCESSOR=x86_64", GLFWPath)
+                            .AssertZeroExitCode();
+                        InheritedShell(build, GLFWPath)
+                            .AssertZeroExitCode();
+                        CopyAll(@out.GlobFiles("src/libglfw.so"), runtimes / "linux-x64" / "native");
+
+                        EnsureCleanDirectory(@out);
+                        
+                        InheritedShell($"{prepare} -DCMAKE_SYSTEM_PROCESSOR=i368", GLFWPath)
```
Comment (Perksey) [inline build/nuke/Build.Native.cs:143]: `i368` doesn't sound right... `i386`?
```suggestion
                        InheritedShell($"{prepare} -DCMAKE_SYSTEM_PROCESSOR=i386", GLFWPath)
```

File: build/nuke/Build.Native.cs [hunk 3]
```diff
@@ -99,4 +99,73 @@ string AndroidHome
...
+                            .AssertZeroExitCode();
+                        InheritedShell(build, GLFWPath)
+                            .AssertZeroExitCode();
+                        
+                        CopyAll(@out.GlobFiles("src/libglfw.so"), runtimes / "linux-x86" / "native");
```
Comment (Perksey) [inline build/nuke/Build.Native.cs:148]: Thinking about it, I think we should just leave out 32-bit linux support given that .NET does not officially support this anyway.
```suggestion
```

File: build/nuke/Build.Support.cs [hunk 4]
```diff
@@ -14,6 +15,10 @@
 using Nuke.Common.CI.GitHubActions;
 using Nuke.Common.IO;
 using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Utilities;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.Tooling.ProcessTasks;
 using Octokit;
 using Octokit.Internal;
```
Comment (Perksey) [inline build/nuke/Build.Support.cs:23]:
```suggestion
using Octokit;
using Octokit.Internal;
using static Nuke.Common.IO.FileSystemTasks;
using static Nuke.Common.Tooling.ProcessTasks;
```

**Review by Perksey — CHANGES_REQUESTED** [review]
Very happy with this!

**Comment by Perksey** [comment]
Oh also... can you change the submodule reference to point at the latest tagged release instead of presumably the latest develop/master/whatever? The latest release SHA is `7d5a16c`.

Thanks!

File: build/nuke/Build.Native.cs (hunk 2 above)
Comment (roeyskoe) [reply build/nuke/Build.Native.cs:143]: Huh, surprised that it did not give any sort of warning/error?

Comment (Perksey) [reply build/nuke/Build.Native.cs:143]: It's probably falling back to a sneaky default.

**Review by Perksey — APPROVED** [review]
Thank you so much - you are a godsend! If the PR description is alluding to you possibly doing the same for OpenAL Soft, feel free to do so!

**Comment by roeyskoe** [comment]
I'll look into it probably Sunday/early next week :)

**Pull request merged**